src/components/bds-switch/test/bds-switch.spec.tsx

```diff
@@ -0,0 +1,103 @@
+import { newSpecPage, SpecPage } from '@stencil/core/testing';
+import { Switch } from '../bds-switch';
+
+describe('bds-switch', () => {
+  const getPage = async (
+    { size, disabled, checked, name } = { size: '', disabled: null, checked: null, name: '' }
+  ): Promise<SpecPage> => {
+    const getSwitchElement = (): string => {
+      if (size != null && size != '' && disabled && checked) {
+        return `<bds-switch name=${name} size=${size} disabled=${disabled} checked=${checked}></bds-switch>`;
+      }
+
+      if (size != null && size != '' && disabled) {
+        return `<bds-switch name=${name} size=${size} disabled=${disabled} ></bds-switch>`;
+      }
+
+      if (size != null && size != '' && checked) {
+        return `<bds-switch name=${name} size=${size}  checked=${checked}></bds-switch>`;
+      }
+
+      if (disabled && checked) {
+        return `<bds-switch  name=${name} disabled=${disabled} checked=${checked}></bds-switch>`;
+      }
+
+      if (size != null && size != '') {
+        return `<bds-switch name=${name} size=${size}></bds-switch>`;
+      }
+
+      if (disabled) {
+        return `<bds-switch name=${name} disabled=${disabled}></bds-switch>`;
+      }
+
+      if (checked) {
+        return `<bds-switch name=${name} checked=${checked}></bds-switch>`;
+      }
+
+      return `<bds-switch name=${name} ></bds-switch>`;
+    };
+
+    return await newSpecPage({
+      html: getSwitchElement(),
+      components: [Switch],
+    });
+  };
+
+  it('should render ', async () => {
+    const page = await getPage();
+
+    expect(page.root.shadowRoot.querySelector('.switch')).toBeTruthy();
+    expect(page.root.shadowRoot.querySelector('.switch--size-standard')).toBeTruthy();
+  });
+
+  it('should render with default props', async () => {
+    const page = await getPage();
+
+    expect(page.root.shadowRoot.querySelector('.switch')).toBeTruthy();
+    expect(page.root.shadowRoot.querySelector('.switch--size-standard')).toBeTruthy();
+    expect(page.root.shadowRoot.querySelector('checked')).toBeFalsy();
```

Comment:
this expect is false-positive, because the ```page.root.shadowRoot.querySelector('checked')``` is returning nothing, you must select the switch and then use the .not and .toHaveProperty('checked') functions, like this:

```
expect(page.root.shadowRoot.querySelector('.switch')).not.toHaveProperty('checked');
```

you can check all the Jest expect functions [here]( https://jestjs.io/docs/pt-BR/expect)